### PR TITLE
Schema.js cleanup

### PR DESCRIPTION
**Pull request by drew-gross**
A collection of semi-random changes that are progress towards database adapters.

### CODE REVIEW

**Comment by flovilmart** [comment]
@drew-gross do you think it's time to move it to `src/Controllers/SchemaController.js`

**Comment by drew-gross** [comment]
Yah I think once these changes are merged I will do that.

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ DatabaseController.prototype.loadSchema = function(acceptor = () => true) {
     }
     this.schemaPromise = this.schemaCollection().then(collection => {
       delete this.schemaPromise;
-      return Schema.load(collection);
+      return Schema.load(collection, this.adapter);
```
Comment (flovilmart) [inline src/Controllers/DatabaseController.js]: we should probably store a SchemaController on `this`

**Comment by flovilmart** [comment]
LGTM, besides the baby comment, for later

File: src/Controllers/DatabaseController.js (hunk 1 above)
Comment (drew-gross) [inline src/Controllers/DatabaseController.js]: Yep, definitely will do something like that in a later PR.

**Pull request merged**